### ISSUE TITLE
Remove position numbers from self-voicing content when moving arm/leg

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/john-travoltage#352]
Discussed during design meeting, these values are useful for a screen reader user but not as much in the self-voicing case. They should be removed when using self-voicing.

**Comment by jessegreenberg** [comment]
I tried this out, but it sounds a little awkward and I am not sure whether to proceed.

When moving the foot, the result is "Leg swing foot off rug." or "Leg swing foot on rug". "Leg swing foot".

For the arm, the output is sometimes something like "Hand not so close to doorknob". But othertimes it is "Hand hand pointing straight up.", because the second "hand" is part of the value description.

@terracoda @Matthew-Moore240 how do you think we should proceed? Do we need to design new value descriptions, or should we leave "position: value" in? Or something else?

**Comment by jessegreenberg** [comment]
Another thing we could try is to use some punctuation to make it sound more natural. Then it could sound like "Leg swing - foot on  rug." or "Hand - hand pointing straight up". I am going to try this really quick.

**Comment by jessegreenberg** [comment]
Yes, adding a ": " added some space between label and value text and it sounds better with the synth. Assigning to @Matthew-Moore240 and @terracoda to review how it sounds on master. Self-voicing is enabled with `?supportsSelfVoicing`

**Comment by terracoda** [comment]
@jessegreenberg, how do I get the version that is not voicing position numbers?

**Comment by terracoda** [comment]
@jessegreenberg, nice job with the colon to create space.
There are a few landmark positions that have "hand" repeated. This is an artifact of describing position. We can remove that repetition:
- Position -13: Hand, Hand pointing away from door, very far from doorknob.
    - Hand, pointing away from door, very far from doorknob.
- Position - 8: Hand, Hand pointing straight up.
    - Hand, pointing straight up.
- Position -4: Hand, Hand pointing at upper door, close to doorknob.
    - Hand, pointing at upper door, close to doorknob.
- Position 4: Hand, Hand pointing to lower door, close to doorknob.
    - Hand, Hand pointing to lower door, close to doorknob.
- Position 7: Hand, Hand pointing straight down.
    - Hand, pointing straight down.
- Position 13: Hand, Hand pointing away from door, very far from doorknob.
    - Hand, pointing away from door, very far from doorknob.

Also, maybe remove "Last stop." from Positions 15 and - 15.

**Comment by terracoda** [comment]
@jessegreenberg, I just realized, I used a comma instead of colon.

Are you using any other punctuation? We could do either:
- Hand, pointing away from door, very far from doorknob. OR
 - Hand: pointing away from door, very far from doorknob.

And not sure we need periods, but it they are there naturally, we can leave them.

**Comment by jessegreenberg** [comment]
I see, sounds good. Yes, it seems that a comma also creates a pause for the synth. I switched it to a comma in the above commit.

**Comment by terracoda** [comment]
@jessegreenberg, I think my https://github.com/phetsims/john-travoltage/issues/352#issuecomment-700999782 was not super clear. I'd like you to Remove the word "Hand" from the object responses at certain locations, so we have no repetition of the word "Hand" during the interaction. Formerly, these 6 positions were like landmarks and with the label as "Hand Position", I felt it was important to repeat the word "Hand" at this landmark position. In Self-Voicing, the label is now "Hand", so the label and value of the where the hand is read nicely together without any repetition.

Here's the list again without the repetition:
 Position -13: 
    - Hand, pointing away from door, very far from doorknob.
- Position - 8: 
    - Hand, pointing straight up.
- Position -4: 
    - Hand, pointing at upper door, close to doorknob.
- Position 4: 
    - Hand, Hand pointing to lower door, close to doorknob.
- Position 7: 
    - Hand, pointing straight down.
- Position 13: 
    - Hand, pointing away from door, very far from doorknob.

Also, I think it might be safe to remove "Last stop" when Hand is "Farthest from door knob."

**Comment by jessegreenberg** [comment]
Sorry, I totally didn't see that comment! OK, understood.

I don't see a good way to reuse existing code/strings while having unique ones just for -15, -13, 8, -4, 4, 7, 13, 15. So I think this will just require new strings/logic for self-voicing.

**Comment by terracoda** [comment]
Oh, that's a bummer.
We should for Quarter see if we want to revisit the design of Interactive Description for John Travoltage.

We might be able to make some re-alignment in the other direction. Pushing some Self-Voicing strings over into the PDOM strings.

**Comment by jessegreenberg** [comment]
Another thing we could try that is more like what we did in GFL:B - instead of reading the appendage label every position change we only read it on focus and when dragging with a mouse. Othwerwise, we just read the position text.

We will still hear "Hand" twice (at some positions) when using mouse/touch or when focus lands on it. But we will never hear duplication when using keyboard.

**Comment by terracoda** [comment]
@jessegreenberg, yes, let's try exactly what you suggested in https://github.com/phetsims/john-travoltage/issues/352#issuecomment-704406029. 

I didn't realize we were doing anything different for the JT sliders. I think that will greatly also improve the Leg Swing, Foot rubbing on rug, and Leg Swing, Foot off rug.

**Comment by jessegreenberg** [comment]
Hmm, I tried to do this but it is turning out to be difficult. in GFLB there was already a difference between keyboard and mouse/touch input so it was easy to make behavior different. That is not true in JT, there isn't currently an easy way to differentiate between keyboard and mouse/touch dragging. 

I think for scalability we would ideally have mouse/keyboard dragging behave the same for self-voicing.

**Comment by jessegreenberg** [comment]
I think the most straight forward way is to either have new strings for self-voicing or change the existing strings in the PDOM. In the aria-valuetext instead of 
"Hand pointing away from door, very far from doorknob"
could we have
"Pointing away from door, very far from doorknob"

?

How important is it to remove "Last stop"? If we want that removed for the self voicing anyway, we will just need to use different strings no matter what.

**Comment by jessegreenberg** [comment]
@jessegreenberg to make the following changes to align self-voicing strings and PDOM strings:

Change "Hand position" -> "Move hand".
Get rid of position numbers.
Replace pdom aria-valuetext with self voicing strings above.
Get rid of "last stop"

Change "Leg swing"
Get rid of leg position numbers

Once done, build a version and assign to @terracoda to review. If it isn't good we will revert and have two sets of strings, one for PDOM, one for self-voicing.

**Comment by terracoda** [comment]
@jessegreenberg, no rush, but please ping me when you have a PDOM (Interactive Description) version I can test.

**Comment by jessegreenberg** [comment]
OK, https://github.com/phetsims/john-travoltage/issues/352#issuecomment-708595270 was done in the above commit. @terracoda would you like to have a listen with an screen reader?

The issue I notice is that without position numbers there are many steps where a user hears the same thing over and over. Like
"Foot off rug", "Foot off rug" Foot off rug" "Foot rubbing on rug". Not sure how large of an issue this is.

https://phet-dev.colorado.edu/html/john-travoltage/1.6.0-dev.28/phet/john-travoltage_en_phet.html

**Comment by terracoda** [comment]
@jessegreenberg, I think with the land marks such as farthest from door know, pointing away from door, pointing straight down, pointing at lower door, pointing at upper door, etc, that the sim sounds great without the position numbers!

I'd like to remove the position numbers from the electron responses, too?
So the new response would be:
- Discharge occurred. Electron on body decreased from {{36}} to {{22}} with hand {{very close to}} doorknob.
- Discharge occurred. Electron on body decreased from {{11}} to {{0}} with hand {{at}} doorknob.

![Screen Shot 2020-10-20 at 1 04 05 PM](https://user-images.githubusercontent.com/1902116/96610953-bf7eed80-12d6-11eb-9561-97b9dd9d07f1.png)
![Screen Shot 2020-10-20 at 1 14 33 PM](https://user-images.githubusercontent.com/1902116/96610971-c60d6500-12d6-11eb-998e-4e7e24bf452e.png)

I just checked the scene/screen summary and the position numbers are either already removed from there, or maybe were never there? Either way, it looks good.
![Screen Shot 2020-10-20 at 1 28 21 PM](https://user-images.githubusercontent.com/1902116/96612338-34065c00-12d8-11eb-9d60-07b8fc1a7739.png)

**Comment by terracoda** [comment]
I think this experience is really grand, @jessegreenberg!

**Comment by terracoda** [comment]
I think the new strings for hand position and leg swing position are safe to use in the PDOM, so I think we should move forward with sharing these strings in the Self-Voicing version.

**Comment by jessegreenberg** [comment]
In the above commit I removed the position value from discharge alert. Then I removed the strings and unused code related to quantitative positions. Closing since that was the only review comment! Will deploy a version shortly to be reviewed with TestFlight.
